pkg/controller/egress/store/egressgroup_test.go

```diff
@@ -0,0 +1,183 @@
+			operations: func(store storage.Interface) {
+				store.Create(egTypeInvalid)
+				store.Create(eg1)
+			},
+			expectedEvents: []watch.Event{
+				{Type: watch.Bookmark, Object: &controlplane.EgressGroup{}},
+				{Type: watch.Added, Object: &controlplane.EgressGroup{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					GroupMembers: nil,
+				}},
+			},
+		},
+		{
+			name:          "Delete event",
+			fieldSelector: fields.SelectorFromSet(fields.Set{"nodeName": "node-local"}),
+			operations: func(store storage.Interface) {
+				store.Create(eg1)
+				store.Update(eg2)
+			},
+			expectedEvents: []watch.Event{
+				{Type: watch.Bookmark, Object: &controlplane.EgressGroup{}},
+				{Type: watch.Added, Object: &controlplane.EgressGroup{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					GroupMembers: nil,
+				}},
+				{Type: watch.Deleted, Object: &controlplane.EgressGroup{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					GroupMembers: nil,
+				}},
+			},
+		},
+		{
+			name:          "Modify event",
+			fieldSelector: fields.SelectorFromSet(fields.Set{"nodeName": "node-local"}),
+			operations: func(store storage.Interface) {
+				store.Create(eg1)
+				store.Update(eg1)
+				store.Update(eg3)
+				store.Update(eg3)
+			},
+			expectedEvents: []watch.Event{
+				{Type: watch.Bookmark, Object: &controlplane.EgressGroup{}},
+				{Type: watch.Added, Object: &controlplane.EgressGroup{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					GroupMembers: nil,
+				}},
+				{Type: watch.Modified, Object: &controlplane.EgressGroupPatch{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					AddedGroupMembers: []controlplane.GroupMember{{Pod: pod1}},
+				}},
+			},
+		},
+		{
+			name:          "Node not Specified",
+			fieldSelector: fields.Everything(),
+			operations: func(store storage.Interface) {
+				store.Create(eg1)
+				store.Update(eg3)
+			},
+			expectedEvents: []watch.Event{
+				{Type: watch.Bookmark, Object: &controlplane.EgressGroup{}},
+				{Type: watch.Added, Object: &controlplane.EgressGroup{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					GroupMembers: nil,
+				}},
+				{Type: watch.Modified, Object: &controlplane.EgressGroupPatch{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					AddedGroupMembers: []controlplane.GroupMember{{Pod: pod1}},
+				}},
+			},
+		},
+	}
+	for _, tc := range tests {
+		store := NewEgressGroupStore()
```

Comment:
Please use subtest to run cases, which makes it easier to figure out which one fails
`t.Run(tc.name, func(t *testing.T) {...}`

Reply:
Updated.